src/esp/core/logging.cpp

```diff
@@ -121,8 +123,14 @@ bool isLevelEnabled(Subsystem subsystem, LoggingLevel level) {
   return level >= LoggingContext::current().levelFor(subsystem);
 }
 
-Cr::Containers::String subsystemPrefix(Subsystem subsystem) {
-  return ""_s.join({"["_s, subsystemNames[uint8_t(subsystem)], "] "_s});
+Cr::Containers::String buildMessagePrefix(Subsystem subsystem,
+                                          const std::string& filename,
+                                          const std::string& function,
+                                          int line) {
+  auto baseFileName = Cr::Utility::Directory::filename(filename);
+  return ""_s.join({"["_s, subsystemNames[uint8_t(subsystem)], "] "_s,
+                    baseFileName, "("_s, std::to_string(line), ")::"_s,
+                    function, " : "_s});
```

Comment:
@jturner65 I was just browsing through some code and came across this -- the following should be significantly more performant, without temporary string allocations for `filename`, `function` and `line` (it returns a `std::string` now instead of a `String`, yes, making that consistent is on my tech debt task pile):

```cpp
std::string buildMessagePrefix(Subsystem subsystem,
                                          const char* filename,
                                          const char* function,
                                          int line) {
  auto baseFileName = Cr::Utility::Directory::filename(filename);
  return Utility::formatString("[{}] {}({})::{} : ", subsystemNames[uint8_t(subsystem)],
                    baseFileName, line, function);
}
```

FYI there's also [CORRADE_FUNCTION](https://doc.magnum.graphics/corrade/Macros_8h.html#a35bb07eaded9ca76d8a2c72135243ded) and [CORRADE_LINE_STRING](https://doc.magnum.graphics/corrade/Macros_8h.html#a074226a9c37ee5816e828a4cf823381d), in case you'd ever need a portable function name or stringified line number.

**Also**, certain terminals and IDEs (KDE's Konsole, VS Code, ...) allow you to click on the filename if you include a reasonable part of the path, and even can jump directly to given line number if you format it as `<filename>:<line>` (which is the format used by GCC messages, GDB, CMake and other tools). So if you want an extra bit of convenience you might want to keep the path and reformat the output slightly. For example this I have in TestSuite, the failure location leads me directly to the line in the editor if I click on it:

![image](https://user-images.githubusercontent.com/344828/133842471-ffb90064-083b-4957-9851-937e537459a3.png)